### PR TITLE
Add discord chatbot workshop

### PR DESCRIPTION
**Pull request by k4u5h4L**
Added the guide to create a chatbot in discord with Discord.js and Dialogflow!

(uses hack club CDN for images)

Conducted on 13 September 2020 by Hack Club NMIT, Bengaluru.

### CODE REVIEW

**Review by sampoder — CHANGES_REQUESTED** [review]
Hey @k4u5h4L! I really like the creativity of this workshop, it certainly looks like fun!

Here are some quick things that would make this workshop epic and (hopefully) ready to go:

- Could we make the workshop a bit shorter? The idea is that Hack Clubbers can build a project within 20 minutes

- Could you please add at least 3 demos of how the project can be expanded on

-  Could you please remove all the additional HTML formatting to make it fit in with the other workshops

- Could you please link a video recording of the test workshop

- Could you also please remove the comments

- Could you please add the metadata to the file 

I think that's about it, thank you!

Excited to see where this goes 🚀

**Comment by k4u5h4L** [comment]
Thank you so much @sampoder for going through my workshop. Yes I will make it a bit shorter and ready to hack within 20 minutes. Also, by video recording of the test workshop you mean a short video of what we got at the end of the workshop right?

**Comment by sampoder** [comment]
> Thank you so much @sampoder for going through my workshop. Yes I will make it a bit shorter and ready to hack within 20 minutes. Also, by video recording of the test workshop you mean a short video of what we got at the end of the workshop right?

Thank you! Yep, just a video of a couple of attendees presenting what they made :)

Thanks mate 🙌

**Comment by k4u5h4L** [comment]
> Thank you! Yep, just a video of a couple of attendees presenting what they made :)

I just wanted to point out that no one presented after that, and the only the recorded workshop is available on YouTube. I had explained everything during the workshop, and I have linked the YouTube video in the workshop. Does that suffice?

Also, can I use `iframe` elements for the video, or should I just link the video URL with it's thumbnail? Because I've noticed that `iframe` doesn't work on GitHub... 

Thank you so much for replying and giving me pointers!

**Comment by sampoder** [comment]
> > Thank you! Yep, just a video of a couple of attendees presenting what they made :)
> 
> I just wanted to point out that no one presented after that, and the only the recorded workshop is available on YouTube. I had explained everything during the workshop, and I have linked the YouTube video in the workshop. Does that suffice?
> 
> Also, can I use `iframe` elements for the video, or should I just link the video URL with it's thumbnail? Because I've noticed that `iframe` doesn't work on GitHub...
> 
> Thank you so much for replying and giving me pointers!

Do you have any pics of what people made, just looking for some proof that it all is understandable.. hope you understand :D

**Comment by sampoder** [comment]
Going to do content review rn

**Review by sampoder — CHANGES_REQUESTED** [review]
Hey! Tried to do the workshop, smth broke :(((

view: https://www.youtube.com/watch?v=vu4jB2upt9M&feature=youtu.be

also has other changes requested so pls make those also please delete all the images from the pr

thank you 🌈!!!

**Comment by k4u5h4L** [comment]
Thanks for leaving a review @sampoder . After watching the video, I realised that Google has changed where the link is located, but I could locate the page required. I will update it in the workshop. Also, I will break down the code into smaller chunks, if people are not able to understand. The reason was to prevent any errors. 

Thanks again for taking your time reviewing my workshop. I will correct those issues asap. Cheers!! :D

**Comment by sampoder** [comment]
Thank you so much!!

**Comment by k4u5h4L** [comment]
Hey @sampoder , I have updated the workshop to support Google's current GCP navigation. And also people have submitted some pictures of them trying out the bot making changes to it. I have updated the same in the `README.md` as well. Hope this is sufficient. Cheers!

p.s. Just tell me how I should share those screenshots with you.

**Comment by sampoder** [comment]
check the workshop bounty blog post for instructions

**Review by lachlanjc — CHANGES_REQUESTED** [review]
Hey, thanks for your submission! Could you please check the box to allow maintainers to edit your file?

- At the beginning of the workshop, please include a screenshot, video, explanation, etc of some kind of what you’re making specifically. The description should also be more specific
- Rename the folder to `discord_chatbot` to be consistent with the rest of the repo
- Check your capitalization in a few places. I immediately noticed the “discord” in the description, an “i” on line 344, “youtube video” on 366, etc
- Please remove the assets folder with images from your PR—just the #cdn links are enough

As part of your submission, you’ll also need to include a video of your conducting of the workshop, as per the bounty guidelines. Thanks!

**Comment by k4u5h4L** [comment]
Thanks for your inputs, I will do these changes as soon as i can. I was (I still am xD) kinda held up with some work. 

Also, I did add a youtube video at the very bottom of us conducting the workshop. Will that suffice?

One last thing, in the beginning I should also add a short video of me describing what we are going to be doing in this workshop right?

Thanks in advance!!

File: workshops/discord_chatbot/README.md
Comment (bezlin6mechminerz) [inline workshops/discord_chatbot/README.md:61]:
```suggestion
Assuming you have logged in to your discord account and made a server or having an existing server as an admin, let's proceed!
```

**Comment by MatthewStanciu** [comment]
Hey, thanks so much for submitting this workshop! I think this is really cool and I would love to see this on the workshops site. However, I feel like there's one big thing that needs to be addressed in this workshop before it's ready to merge: it says it takes 20 minutes at the beginning, but the demo you linked at the bottom of the workshop takes an hour to get through, and the workshop itself is very long. I think some of it should be shortened a little bit, and you should make sure it actually takes 20 minutes to get through entirely.

Also, I think you should move the video link out of your workshop and instead paste the YouTube link into the #workshop-skunkworks channel in Slack.

I think this workshop is super cool and I would love it if you resubmitted with it being a little shorter than it currently is. Ping me on Slack if you need any guidance. Looking forward to your resubmission!

**Comment by k4u5h4L** [comment]
@MatthewStanciu thank you so much for replying! The main reason the workshop was long was because I was explaining the code for more time as there may be people who aren't well versed in JavaScript yet. But if the length of the video is an issue, then sure, I will try and cut down on the time and fit it in the given time frame.

Again, thanks for giving inputs, I will be resubmitting the PR again with the needed changes!

**Pull request closed without merging**
